### PR TITLE
Feat/service tab

### PR DESCRIPTION
**Pull request by RealGoodProgrammer**
image

### CODE REVIEW

File: src/components/nodes/components/NodesTableHead.vue [hunk 1]
```diff
@@ -2,7 +2,9 @@
   <thead :class="classes.root">
     <tr>
       <th :class="classes.checkbox" v-if="!hideCheckbox" />
-      <th :class="classes.label" class="pl-0 pr-2" v-if="!hideLabel">{{ t('nodes.coin') }}</th>
+      <th :class="classes.label" class="pl-0 pr-2" v-if="!hideLabel">
+        {{ customLabelCode ? t(customLabelCode) : t('nodes.coin') }}
+      </th>
```
Comment (bludnic) [inline src/components/nodes/components/NodesTableHead.vue:7]:
```suggestion
      <th :class="classes.label" class="pl-0 pr-2" v-if="!hideLabel">
        {{ label }}
      </th>
```

1. Rename prop to `label`
2. Pass it already translated, e.g. `<NodesTableHead :label="$t('nodes.service')" />`
3. Pass the label explicitly for "ADM" and "Coins tab", e.g. `<NodesTableHead :label="$t('nodes.coin')" />`

File: src/store/modules/rate/rate-actions.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            commit('setRates', result)
+          }
+          commit('loadRates')
+          resolve(result)
+        })
+        .catch((err) => {
+          reject(err)
+        })
+    })
+  },
```
Comment (bludnic) [inline src/store/modules/rate/rate-actions.ts:26]: There no need in wrapping in `new Promise()`.

1. Prefix function with `async`

2. Use as:

```js
const { result, success } = await rateInfoClient.getAllRates()
if (success) {
  commit('setRates', result)
}
```

Comment (bludnic) [reply src/store/modules/rate/rate-actions.ts:26]: Similar for `getHistoryRates`

File: src/store/modules/rate/rate-mutations.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { Rates, RateState } from '@/store/modules/rate/types.ts'
+
+export const mutations: MutationTree<RateState> = {
+  setRates(state, rates: Rates) {
+    state.rates = rates
+  },
+  setHistoryRates(state, historyRates: { name: number; value: Rates }) {
+    state.historyRates[historyRates.name] = historyRates.value
+  },
+  loadRates(state) {
```
Comment (bludnic) [inline src/store/modules/rate/rate-mutations.ts:11]: Use better name, e.g. `setLoaded`.

File: src/store/modules/services/services-mutations.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { MutationTree } from 'vuex'
+import { AvailableService, ServicesState } from '@/store/modules/services/types.ts'
+
+export const mutations: MutationTree<ServicesState> = {
+  useFastestService(state, value) {
+    state.useFastestService = value
+  },
+
+  toggle(state, payload: { type: AvailableService; url: string; active: boolean }) {
```
Comment (bludnic) [inline src/store/modules/services/services-mutations.ts:9]: I'd suggest to define types for payload on the top of file.

type TogglePayload = { type: AvailableService; url: string; active: boolean }

File: src/store/modules/services/services-mutations.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      node.active = payload.active
+    }
+  },
+
+  status(
+    state,
+    {
+      status,
+      serviceType
+    }: { serviceType: AvailableService; status: { url: string; active: boolean } }
```
Comment (bludnic) [inline src/store/modules/services/services-mutations.ts:24]: Move type at the top of the file `type StatusPayload`. See previous comment.

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/lib/nodes/services.ts [hunk 6]
```diff
@@ -1,7 +1,9 @@
 import { klyIndexer } from './kly-indexer'
 import { ethIndexer } from './eth-indexer'
+import { rateInfoClient } from './rate-info-service'
 
 export const services = {
   klyIndexer,
-  ethIndexer
+  ethIndexer,
+  'rates-info': rateInfoClient
```
Comment (bludnic) [inline src/lib/nodes/services.ts:8]:
```suggestion
  ratesInfo: rateInfoClient
```

Better to follow same camelCase style

**Review by PaulDremanovich — APPROVED** [review]

File: src/lib/nodes/rate-info-service/RateInfoService.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+export type RateHistoryInfoResponse = {
+  success: boolean
+  date: number
+  result: {
+    _id: string
+    date: number
+    tickers: Record<string, number>
+  }[]
+}
```
Comment (bludnic) [inline src/lib/nodes/rate-info-service/RateInfoService.ts:20]: These types belong to `RateInfoClient`. Move them to `/lib/nodes/rate-info-service/types/`

File: src/components/nodes/services/ServiceNodesTableItem.vue [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+      <NodeStatus :node="node" />
+    </NodeColumn>
+  </tr>
+</template>
+
+<script lang="ts">
+import { computed, PropType } from 'vue'
+import { useStore } from 'vuex'
+import type { NodeStatusResult } from '@/lib/nodes/abstract.node.ts'
+import type { TNodeLabel } from '@/lib/nodes/constants.ts'
```
Comment (bludnic) [inline src/components/nodes/services/ServiceNodesTableItem.vue:25]: Remove `.ts` extension from imports. Check other files as well.

Comment (bludnic) [reply src/components/nodes/services/ServiceNodesTableItem.vue:25]: Use relative import where possible.

File: src/lib/nodes/rate-info-service/RateInfoClient.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  constructor(endpoints: string[] = [], minNodeVersion = '0.0.0') {
+    super('adm', 'service')
+    this.nodes = endpoints.map((endpoint) => new RateInfoService(endpoint))
+    this.minNodeVersion = minNodeVersion
+
+    void this.watchNodeStatusChange()
+  }
+
+  async getAllRates(): Promise<RateInfoResponse> {
+    const node = await this.fetchNode()
```
Comment (bludnic) [inline src/lib/nodes/rate-info-service/RateInfoClient.ts:14]: I think you can reuse `getNode()` from parent class and remove `async fetchNode()` below.

```suggestion
    const node = this.getNode()
```

File: src/lib/nodes/rate-info-service/RateInfoClient.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    return await node.getAllRates()
+  }
+
+  async getHistory(timestamp: number) {
+    const node = await this.fetchNode()
```
Comment (bludnic) [inline src/lib/nodes/rate-info-service/RateInfoClient.ts:19]: Check the comment above

```suggestion
    const node = this.getNode()
```

File: src/lib/nodes/rate-info-service/RateInfoService.ts [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  async getAllRates(): Promise<RateInfoResponse> {
+    const response = await this.client.get<RateInfoResponse>('/get')
+    return response.data
+  }
+
+  async getHistory(timestamp: number) {
+    const response = await this.client.get<RateHistoryInfoResponse>(
+      `/getHistory?timestamp=${timestamp}`
```
Comment (bludnic) [inline src/lib/nodes/rate-info-service/RateInfoService.ts:39]: Pass query params using axios `params` prop. Add TS type `type GetHistoryParams`

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/store/modules/services/services-plugin.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import { services } from '@/lib/nodes/services'
+import { Store } from 'vuex'
+import { AvailableService, ServicesState } from '@/store/modules/services/types'
+import { NODE_LABELS } from '@/lib/nodes/constants'
+
+export default (store: Store<ServicesState>) => {
```
Comment (bludnic) [inline src/store/modules/services/services-plugin.ts:6]:
```ts
import { Plugin } from 'vuex'

export default const servicesPlugin: Plugin<ServicesState> = (store) => ...
```

**Review by bludnic — CHANGES_REQUESTED** [review]

**Comment by PaulDremanovich** [comment]
[Add health check for indexers](https://trello.com/c/DDOcYSIx/496-add-health-check-for-indexers)

**Comment by bludnic** [comment]
[Add infoService health check](https://trello.com/c/tpbSZXFE/482-add-infoservice-health-check)

**Review by bludnic — APPROVED** [review]

**Pull request merged**